src/ipc.c

```diff
@@ -119,12 +119,11 @@ umf_result_t umfPutIPCHandle(umf_ipc_handle_t umfIPCHandle) {
     return ret;
 }
 
-umf_result_t umfOpenIPCHandle(umf_memory_pool_handle_t hPool,
+umf_result_t umfOpenIPCHandle(umf_ipc_handler_handle_t hIPCHandler,
                               umf_ipc_handle_t umfIPCHandle, void **ptr) {
 
-    // We cannot use umfPoolGetMemoryProvider function because it returns
-    // upstream provider but we need tracking one
-    umf_memory_provider_handle_t hProvider = hPool->provider;
+    // IPC handler is an instance of tracking memory provider
```

Comment:
Can we add check  `(uint32_t *)hIPCHandler == UMF_VERSION` here?

provider structure has version on first field. Pool struct has pool priv. So in this way we can detect that user provided IPChandler, not a pool ptr. This might be useful, as this function, in fact does not change from ABI - it still ()(void *, void *, void**), so in that way we can detect if user did something wrong, and give better error message. 

We could later on remove this check, but for first version with different api it might be useful

Reply:
I understand the trick you suggest to implement. You want to catch the case when a user still passes the pool handle as an argument. 

But it looks like a hack, should we use it in a situation when we have only 3 internal customers who are using this API so far? We can just negotiate with them the required change.

Reply:
Ok, Done.